posts/2024-11-25-cloudant-with-open-liberty.adoc

```diff
@@ -0,0 +1,170 @@
+
+[source,java]
+----
+@Produces
+public Cloudant createCloudant() {
+     String password = PasswordUtil.passwordDecode(encodedPassword);
+     BasicAuthenticator authenticator = new BasicAuthenticator.Builder()
+                .username(username)
+                .password(password)
+                .build();
+
+    return new Cloudant("cloudant", authenticator);
+}
+----
+This requires `passwordUtilities-1.0`, available as a Maven dependency:
+[source,xml]
+----
+<dependency>
+    <groupId>io.openliberty.features</groupId>
+    <artifactId>passwordUtilities-1.0</artifactId>
+    <version>18.0.0.4</version>
+</dependency>
+----
+
+You also need to enable the feature in `server.xml`:
+[source, xml]
+----
+<feature>passwordUtilities-1.0</feature>
+----
+
+By default, Cloudant communication occurs over HTTPS (SSL). However, if you're working with a custom Cloudant instance or self-signed certificates, you may need to configure your Open Liberty application to trust the custom certificate.
+
+You can configure SSL by modifying the server.xml to use your custom SSL certificates:
+[source, xml]
+----
+<sslRegistry id="defaultSSLConfig">
+    <keyStore file="path/to/keystore.jks" password="password"/>
+    <trustStore file="path/to/truststore.jks" password="password"/>
+</sslRegistry>
+
+<httpEndpoint host="*" httpPort="9080" httpsPort="9443" sslConfigRef="defaultSSLConfig"/>
+----
+
+== Configure with MicroProfile Config
+Now that you have a more advanced CDI producer, it would be nice to add configurability to some of the variables, like the user and password. Using link:{url-prefix}/guides/microprofile-config-intro.html[MicroProfile Config] makes configuring the Cloudant driver simple. You can add the following to your CDI producer to add configuration:
+[source, java]
+----
+@Inject
+ @ConfigProperty(name = "cloudant.host", defaultValue = "localhost")
+ String host;
+
+ @Inject
+ @ConfigProperty(name = "cloudant.port", defaultValue = "5984")
+ String port;
+
+ @Inject
+ @ConfigProperty(name = "cloudant.username")
+ String username;
+
+ @Inject
+ @ConfigProperty(name = "cloudant.password")
+ String password;
+
+ @Inject
+ @ConfigProperty(name = "cloudant.dbname")
+ String dbname;
+----
+Now, by placing the following snippet in your ``microprofile-config.properties`` or `server.env` file, the values for user and password will be pulled into the CloudantProducer class:
+[source, text]
+----
+cloudant.user=admin
+cloudant.password={aes}AEEjCqvh7XAwDxrdYC6BUbqYlwqI8NAxRkWWWq7muxZu
+----
+== No need for a Cloudant feature
+Previously, using Cloudant required enabling the `cloudant-1.0` feature.  Even if the Cloudant Java Driver API changes, simple updates to your CDI producer will allow it to continue to work. You should remove the `cloudant-1.0` feature from your `server.xml` when using newer versions of Cloudant with a CDI producer.
+
+The Cloudant client should be bundled in your application. To do this with Maven you can use a dependency:
```

Comment:
The Cloudant SDK for Java should be bundled in your application. To do this with Maven you can use a dependency: